soccer/planning/planner/SettlePlanner.cpp

```diff
@@ -0,0 +1,177 @@
+#include "SettlePlanner.hpp"
+#include "motion/TrapezoidalMotion.hpp"
+#include "planning/trajectory/RRTUtil.hpp"
+#include "planning/trajectory/Trajectory.hpp"
+#include "planning/trajectory/VelocityProfiling.hpp"
+
+namespace Planning {
+ConfigDouble* SettlePlanner::_bufferTimeBeforeContact;
+ConfigDouble* SettlePlanner::_targetPointGain;
+ConfigDouble* SettlePlanner::_searchStartDist;
+ConfigDouble* SettlePlanner::_searchEndDist;
+ConfigDouble* SettlePlanner::_searchIncDist;
+ConfigDouble* SettlePlanner::_targetChangeThreshold;
+ConfigDouble* SettlePlanner::_shortcutDist;
+
+REGISTER_CONFIGURABLE(SettlePlanner);
+
+void SettlePlanner::createConfiguration(Configuration* cfg) {
+    _bufferTimeBeforeContact =
+        new ConfigDouble(cfg, "Capture/Settle/bufferTimeBeforeContact");
+    _targetPointGain = new ConfigDouble(cfg, "Capture/Settle/targetPointGain");
+    _searchStartDist = new ConfigDouble(cfg, "Capture/Settle/searchStartDist");
+    _searchEndDist = new ConfigDouble(cfg, "Capture/Settle/searchEndDist");
+    _searchIncDist = new ConfigDouble(cfg, "Capture/Settle/searchIncDist");
+    _targetChangeThreshold =
+        new ConfigDouble(cfg, "Capture/Settle/targetChangeThreshold");
+    _shortcutDist = new ConfigDouble(cfg, "Capture/Settle/shortcutDist");
+}
+using namespace Geometry2d;
+// todo(motion planning) find a better way to handle moving targets
+RJ::Time SettlePlanner::bruteForceIntercept(const PlanRequest& request) {
+    Ball& ball = request.context->state.ball;
+    std::vector<double> dists;
+    double endDist = *_searchEndDist;
+    const Rect& fieldRect = Field_Dimensions::Current_Dimensions.FieldRect();
+    Segment ballLine{ball.pos, ball.pos + ball.vel.normalized(endDist)};
```

Comment:
The actual velocity is fairly noisy as well. #1233 has some graphs of velocity vs time as I roll the ball across the field. This will make the target point fairly noisy because of time estimate.